src/thread.ts

```diff
@@ -401,33 +382,47 @@ export class Thread<SCG extends ExtendableGenerics = DefaultGenerics> {
     }
   };
 
-  public markAsRead = () => {
-    const { read } = this.state.getLatestValue();
-    const currentUserId = this.client.user?.id;
-    const { unreadMessageCount } = (currentUserId && read[currentUserId]) || {};
-    if (!unreadMessageCount) return;
-    return this.channel.markRead({ thread_id: this.id });
+  public markAsRead = async () => {
+    const currentUserId = this.client.userID;
+
+    if (currentUserId && this.ownUnreadCount) {
+      const currentUserRead = this.state.getLatestValue().read[currentUserId];
+
+      if (currentUserRead) {
+        this.state.next((current) => ({
+          ...current,
+          read: {
+            ...current.read,
+            [currentUserId]: {
+              ...currentUserRead,
+              unreadMessageCount: 0,
+            },
+          },
+        }));
+      }
```

Comment:
Is this an optimistic update of sorts? We're reacting to `message.read` event to reset this and it's such a value where optimistic update could be omitted in my opinion.

Reply:
The optimistic update is useful for active threads. It marks an active thread read immediately, which I think is nicer than having a short period when an active thread is unread.

Reply:
Let's add a TODO then to handle reset to a previous state in case `markRead` request fails. I think it's highly unlikely but a possibility nonetheless.

Reply:
Fixed in https://github.com/GetStream/stream-chat-js/pull/1347/commits/805ad1809e5650c1a40059cb90e38ac19ac55641.

1. `markAsRead` no longer optimistically updates read state for current user.
2. We don't count new replies in an active thread towards unread count of the current user.
3. `markAsRead` calls in subscriptions are throttled.